boilerplate/.dependency-cruiser.js

```diff
@@ -0,0 +1,207 @@
+        "If there's something in a spec that's of use to other modules, it doesn't have that single " +
+        "responsibility anymore. Factor it out into (e.g.) a separate utility/ helper or a mock.",
+      severity: "error",
+      from: {
+        pathNot: "\\.(spec|test)\\.(js|mjs|cjs|ts|tsx)$",
+      },
+      to: {
+        path: "\\.(spec|test)\\.(js|mjs|cjs|ts|tsx)$",
+      },
+    },
+    {
+      name: "not-to-dev-dep",
+      severity: "error",
+      comment:
+        "This module depends on an npm package from the 'devDependencies' section of your " +
+        "package.json. It looks like something that ships to production, though. To prevent problems " +
+        "with npm packages that aren't there on production declare it (only!) in the 'dependencies'" +
+        "section of your package.json. If this module is development only - add it to the " +
+        "from.pathNot re of the not-to-dev-dep rule in the dependency-cruiser configuration",
+      from: {
+        path: "^(app|src)",
+        pathNot: "\\.(spec|test)\\.(js|mjs|cjs|ts|tsx)$",
+      },
+      to: {
+        dependencyTypes: ["npm-dev"],
+        pathNot: ["node_modules/@types/"],
+        exoticRequireNot: [
+          "react-native/Libraries/Utilities/codegenNativeComponent",
+          "react-native/Libraries/Utilities/codegenNativeCommands",
+        ],
+      },
+    },
+    {
+      name: "optional-deps-used",
+      severity: "info",
+      comment:
+        "This module depends on an npm package that is declared as an optional dependency " +
+        "in your package.json. As this makes sense in limited situations only, it's flagged here. " +
+        "If you're using an optional dependency here by design - add an exception to your" +
+        "dependency-cruiser configuration.",
+      from: {},
+      to: {
+        dependencyTypes: ["npm-optional"],
+      },
+    },
+    {
+      name: "peer-deps-used",
+      comment:
+        "This module depends on an npm package that is declared as a peer dependency " +
+        "in your package.json. This makes sense if your package is e.g. a plugin, but in " +
+        "other cases - maybe not so much. If the use of a peer dependency is intentional " +
+        "add an exception to your dependency-cruiser configuration.",
+      severity: "warn",
+      from: {},
+      to: {
+        dependencyTypes: ["npm-peer"],
+      },
+    },
+  ],
+  options: {
+    doNotFollow: {
+      path: "node_modules",
+    },
+    tsPreCompilationDeps: true,
+    tsConfig: {
+      fileName: "tsconfig.json",
+    },
+    enhancedResolveOptions: {
+      exportsFields: ["exports"],
+      conditionNames: ["import", "require", "node", "default"],
+      // React Native / Metro bundler support for platform-specific extensions
+      // See: https://reactnative.dev/docs/platform-specific-code
+      // See: https://github.com/sverweij/dependency-cruiser/issues/511
+      extensions: [
+        ".ios.tsx",
+        ".android.tsx",
```

Comment:
Do we need .web.t|jsx? extensions here?

Reply:
It seems that expo has these two platforms manually set 🤔 https://github.com/expo/expo/blob/main/packages/%40expo/metro-config/src/ExpoMetroConfig.ts#L299

Reply:
Yeah I just figured it would be good coverage, if you think otherwise that's fine. Trying to think how it would scale